textdb/textdb-api/src/test/java/edu/uci/ics/textdb/api/common/SchemaTest.java

```diff
@@ -43,4 +42,22 @@ public void testGetIndex(){
         Assert.assertEquals(expectedIndex_2, retrievedIndex_2);
         
     }
+    
+    @Test(expected = UnsupportedOperationException.class)
+    public void testAddingNewAttribute(){ //Should fail

```

Comment:
@chenlica The initial design by @JavierJia was to make the `ITuple` object immutable, so that, the `ITuple` objects are passed to different layers without expecting any changes to them. Since `Schema` is also part of the `ITuple` object I made it immutable.
But storing the fields as `Array` object (or `Arrays.asList()`) doesn't make it completely immutable, since we can replace the element at particular index. We can make it completely immutable by using any third party `ImmutableList` (Java doesn't provide it by default). I will let @JavierJia to make decision on this.